### PR TITLE
Topic/server unresponsive

### PR DESCRIPTION
**Pull request by crucialfelix**
This cherry picks the Server.unresponsive and the new ServerStatusWatcher class refactoring from https://github.com/supercollider/supercollider/pull/1655

It does not include the Recording or Volume features from there. That PR was far too big.
There are also some help file updates and way too much reformatting in that PR.

### CODE REVIEW

**Comment by crucialfelix** [comment]
There are problems with extra variables and methods. Do not merge yet

**Comment by telephon** [comment]
thanks for that. I hope it is not an endless task to pick all those apart.

**Comment by crucialfelix** [comment]
it is a bit, but I like a challenge.I just decided I couldn't merge all of that. 

too many chances to introduce bugs. far too many files affected

also github now offers the ability to squash the commits when merging.

**Comment by telephon** [comment]
I guess if I had squashed some in the first place, it would have been more compact. It is quite a bit of a process the whole thing. Thanks!

**Comment by crucialfelix** [comment]
personally I've never been fond of "git interactive" squashing. I should
learn to use that more.

On Mon, Apr 4, 2016 at 11:30 PM Julian Rohrhuber <notifications@github.com>
wrote:

> I guess if I had squashed some in the first place, it would have been more
> compact. It is quite a bit of a process the whole thing. Thanks!
>
> —
> You are receiving this because you authored the thread.
>
> Reply to this email directly or view it on GitHub
> <https://github.com/supercollider/supercollider/pull/1935#issuecomment-205504736>
>
-- 
https://twitter.com/crucialfelix
https://medium.com/@crucialfelix
http://www.mattermind.com/

**Comment by miguel-negrao** [comment]
On 04-04-2016 22:40, Chris Sattinger wrote:
> personally I've never been fond of "git interactive" squashing. I should
> learn to use that more.
>
It is really usefull !! While working on a feature I will commit often,
sometimes just before I introduce dangerous changes so I can revert, or
a the end of the day, so the history is a mess. Then when I'm done I can
either squash and rename some commits or even 'reset' the whole thing
(so put all changes back in the index) and then do the commits from
scratch dividing the changes in a nice way for posterity. It also helps
that the program I use, smartgit, has a very intuitive gui for git
interactive rebasing, it just shows all commits not pushed yet to the
remote and you can reorder them, reword commits and squash them, all
from the gui.

Best,
Miguel

**Comment by bagong** [comment]
What throws me off: you have several change in the same file that logically belong to different commits. Is there a way to distinguish different changes in the same file?

File: SCClassLibrary/Common/Control/Server.sc [hunk 1]
```diff
@@ -669,77 +577,91 @@ Server {
...
+	aliveThreadPeriod_ { |val|
+		statusWatcher.aliveThreadPeriod_(val)
+	}
+	aliveThreadPeriod { |val|
+		^statusWatcher.aliveThreadPeriod
 	}
+
+
+
+	disconnectSharedMemory {
```
Comment (crucialfelix) [inline SCClassLibrary/Common/Control/Server.sc]: this wasn't supposed to be here. I will add a commit to remove it. its part of some other feature

**Comment by gusano** [comment]
> What throws me off: you have several change in the same file that logically belong to different commits

@bagong I guess it's a matter of refraining oneself to *not commit* before starting to write unrelated changes to a file.

**Comment by crucialfelix** [comment]
or use a git gui that lets you select just the lines you want to commit.
that's what I always do.

http://gitup.co is great

magit in emacs,

sourcetree is still what I use the most,

gitkraken has the best name and color scheme
https://www.gitkraken.com/


On Tue, Apr 5, 2016 at 5:58 PM Yvan Volochine <notifications@github.com>
wrote:

> What throws me off: you have several change in the same file that
> logically belong to different commits
>
> @bagong <https://github.com/bagong> I guess it's a matter of refraining
> oneself to *not commit* before starting to write unrelated changes to a
> file.
>
> —
> You are receiving this because you authored the thread.
>
> Reply to this email directly or view it on GitHub
> <https://github.com/supercollider/supercollider/pull/1935#issuecomment-205869314>
>
-- 
https://twitter.com/crucialfelix
https://medium.com/@crucialfelix
http://www.mattermind.com/

**Comment by gusano** [comment]
> or use a git gui

[OT] I'd suggest properly learning `git (cli)` instead, it's not that hard and kinda forces one to do things "the-right-way" :japanese_ogre: 

ps @crucialfelix It would be great if you could strip off the quoted messages when replying to issue/PR comments via email.. Would make it easier to read on the website :)

**Comment by miguel-negrao** [comment]
> or use a git gui
> 
> [OT] I'd suggest properly learning git (cli) instead, it's not that hard and kinda forces one to do things "the-right-way" :japanese_ogre:

Totally agree in general, but actually for this particular feature I think doing it in the cli is really a p.i.t.a. ... :-) Specifying line numbers is so much easier by just selecting them with a mouse.

**Comment by bagong** [comment]
Yea, I'll have a better look at gui tools. Never used them much except for comparing branches.

File: SCClassLibrary/Common/Control/Server.sc (hunk 1 above)
Comment (crucialfelix) [inline SCClassLibrary/Common/Control/Server.sc]: it is supposed to be there. tested, no worries.

**Comment by telephon** [comment]
Thanks for the cherry picking labour!

**Comment by timblechmann** [comment]
fwiw, the ide uses spaces, not tabs. please keep the style consistent, mixing tabs and spaces is a pain to work with.

**Comment by crucialfelix** [comment]
let's set up a travis check that will block any PR that aren't correct. there is probably a full style check utility for c++ (like the fabulous jcsc for javascript)

of course first we will have to cleanup all the code files.

**Comment by vivid-synth** [comment]
@crucialfelix maybe we don't have to clean up existing code; we just block any incorrect additions?

**Comment by crucialfelix** [comment]
any new PR would scold the submitter for all the sins of previous
generations.
the style checker would have to check the whole code base, not just what is
being submitted.

I've used cpplint before, but AFAICT you cannot customize it. it uses
google's style guide.


On Fri, May 6, 2016 at 12:40 AM vivid-synth <notifications@github.com>
wrote:

> @crucialfelix <https://github.com/crucialfelix> maybe we don't have to
> clean up existing code; we just block any incorrect additions?
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/supercollider/supercollider/pull/1935#issuecomment-217301017>
>
-- 
https://twitter.com/crucialfelix
https://medium.com/@crucialfelix
http://www.mattermind.com/

**Comment by timblechmann** [comment]
for sc-ide or supernova, just grepping for tabs should be a good start. clang-format is promising, but i wasn't too happy with the results when i tried it with supernova, as it still has some limitations mainly regarding alignment of identifiers. might have improved since that time, though.

**Pull request merged**